### PR TITLE
Port FunctionalTests to .NET Core 3

### PR DESCRIPTION
**Pull request by bricelam**
Tests were excluded for one of three reasons:
- No SqlServerCe
- No Spatial
- No CodeDOM

Part of #271

### CODE REVIEW

File: src/EntityFramework/Migrations/Model/ColumnModel.cs [hunk 1]
```diff
@@ -52,31 +51,6 @@ public ColumnModel(PrimitiveTypeKind type, TypeUsage typeUsage)
             : base(type, typeUsage)
         {
             _clrType = PrimitiveType.GetEdmPrimitiveType(type).ClrEquivalentType;
-            _clrDefaultValue = CreateDefaultValue();
```
Comment (bricelam) [inline src/EntityFramework/Migrations/Model/ColumnModel.cs]: Made this lazy to enable model diffing in the tests to detect schema changes

File: test/EntityFramework/FunctionalTests.ProviderAgnostic/CodeFirst/InheritanceScenarioTests.cs [hunk 2]
```diff
@@ -11,11 +11,7 @@ namespace System.Data.Entity.CodeFirst
 
     public class InheritanceScenarioTests
     {
-        [Fact(
-#if NETCOREAPP3_0
-            Skip = "dotnet/corefx#37849"
```
Comment (bricelam) [inline test/EntityFramework/FunctionalTests.ProviderAgnostic/CodeFirst/InheritanceScenarioTests.cs]: Fixed! (Great turnaround.)

File: test/EntityFramework/FunctionalTests/Migrations/AutoAndGenerateScenarios.cs [hunk 3]
```diff
@@ -1147,7 +1148,7 @@ public class V2 : AutoAndGenerateContext_v2
         {
             protected override void OnModelCreating(DbModelBuilder modelBuilder)
             {
-                if (Database.Connection is SqlCeConnection)
+                if (this.IsSqlCe())
```
Comment (bricelam) [inline test/EntityFramework/FunctionalTests/Migrations/AutoAndGenerateScenarios.cs]: Oops. Can revert.

File: test/EntityFramework/FunctionalTests/ProductivityApi/SpatialTests.cs [hunk 4]
```diff
@@ -38,8 +40,6 @@ static SpatialTests()
             _connectionString = String.Format(
                 CultureInfo.InvariantCulture, baseConnectionString,
                 SimpleConnectionString<SpatialNorthwindContext>());
-
-            SqlServerTypes.Utilities.LoadNativeAssemblies(AppDomain.CurrentDomain.BaseDirectory);
```
Comment (bricelam) [inline test/EntityFramework/FunctionalTests/ProductivityApi/SpatialTests.cs]: Called in base. (Forgot to remove.)

File: test/EntityFramework/FunctionalTests/Query/LinqToEntities/StringConcatTests.cs [hunk 5]
```diff
@@ -194,7 +194,7 @@ public void Issue1904_StringConcatPlus_can_concat_constant_string_with_property_
                     Assert.Equal("a" + projection.Entity.BoolProp + "b", projection.BoolProp);
                     Assert.Equal("a" + projection.Entity.GuidProp + "b", projection.GuidProp);
                     Assert.Equal("a" + projection.Entity.FloatProp.ToString(NumberFormatInfo.InvariantInfo) + "b", projection.FloatProp);
-                    Assert.Equal("a" + projection.Entity.DoubleProp.ToString(NumberFormatInfo.InvariantInfo) + "b", projection.DoubleProp);
+                    Assert.Equal("a" + projection.Entity.DoubleProp.ToString("G15", NumberFormatInfo.InvariantInfo) + "b", projection.DoubleProp);
```
Comment (bricelam) [inline test/EntityFramework/FunctionalTests/Query/LinqToEntities/StringConcatTests.cs]: I wonder if SQL Server will ever take the breaking change...

File: test/EntityFramework/FunctionalTests/WrappingProvider/WrappingProviderTests.cs [hunk 6]
```diff
@@ -336,6 +339,10 @@ private static void RegisterAdoNetProvider(Type providerFactoryType)
 
             _providerTable.Rows.Remove(_providerTable.Rows.Find(SqlClientInvariantName));
             _providerTable.Rows.Add(row);
+#else
+            DbProviderFactories.UnregisterFactory(SqlClientInvariantName);
+            DbProviderFactories.RegisterFactory(SqlClientInvariantName, providerFactoryType.AssemblyQualifiedName);
```
Comment (bricelam) [inline test/EntityFramework/FunctionalTests/WrappingProvider/WrappingProviderTests.cs]: ❤️ Thanks, Frans!

File: test/EntityFramework/FunctionalTests/Query/LinqToEntities/StringConcatTests.cs (hunk 5 above)
Comment (ajcvickers) [reply test/EntityFramework/FunctionalTests/Query/LinqToEntities/StringConcatTests.cs]: No comment. ;-)

**Pull request merged**